src/UI/Components/Diagram/helpers.ts

```diff
@@ -929,3 +904,52 @@ export function showLinkTools(
 
   linkView.addTools(tools);
 }
+
+/**
+ * Function that remove data about inter-service relation between cells
+ * @param {dia.Paper} paper JointJS paper object
+ * @param {dia.Graph} graph JointJS graph object
+ * @param {ServiceEntityBlock} elementCell cell that we checking
+ * @param {ServiceEntityBlock} disconnectingCell cell that is being connected to elementCell
+ * @returns {void}
+ */
+const removeConnectionData = (
+  paper: dia.Paper,
+  graph: dia.Graph,
+  elementCell: ServiceEntityBlock,
+  disconnectingCell: ServiceEntityBlock,
+): void => {
+  const elementRelations = elementCell.getRelations();
+
+  // resolve any possible relation connections between cells
+  if (elementRelations && elementRelations.has(String(disconnectingCell.id))) {
+    elementCell.removeRelation(String(disconnectingCell.id));
+
+    //get all connected entities to inter-service relation cell (the current relation that will be removed is included here)
+    const disconnectingCellNeighbors = graph.getNeighbors(disconnectingCell);
+
+    //the inters-service relataion cell can be also embedded entity, so attempt to get id of the holder
+    const embeddedToID = disconnectingCell.get("embeddedTo");
+
+    //filter out every cell that is either embedded entity of to the disconnectingCell or is core or the disconnectingCell is embedded to it
+    const interServiceRelations = disconnectingCellNeighbors.filter(
+      (cell) =>
+        !(cell.get("embeddedTo") === disconnectingCell.id) ||
+        !(embeddedToID === cell.id),
+    );
+
+    //all left are the inter-service relation connections, if there is more than one then we should highlight the cell
```

Comment:
the choice of name of the variable, and the comment above make it hard to follow why having more than one element should highlight the cell. I also find it odd you're using `=== 1 ` then just below 🤔 and not `< 1`

Reply:
improved comment, should be, if there is only one cell - as we are including the one that we are trying to disconnect. so 0 should not be possible here, if there is more than 1, which can theoretically happen, then we don't highlight that cell as it's already connected